### PR TITLE
Synthanol v1.1

### PR DESCRIPTION
**Pull request by FlattestGuitar**
Synthanol will now notify organic users of how awfully it tastes and make them puke. No toxin damage or anything of the sort. I wasn't sure how to get rid of it, so it will also make organics slur and get confus if they drink enough, *but shhh, that's a secret.*

Also adds a few new IPC drinks and makes Holy Water have its own glass (uses the boring sprite of a water glass though).

MOAR BOOZE
:cl:
add: Adds three new synthanol drinks
tweak: Drinking synthanol is now a bad idea if you're organic
tweak: A glass of holy water is now identifiable
/:cl:

Also, don't judge those commits. I _really_ suck at GitHub.

### CODE REVIEW

File: code/modules/reagents/newchem/drinks.dm [hunk 1]
```diff
@@ -93,16 +94,32 @@
...
-	if(d >= collapse_start && prob(10))
-		M.emote("collapse")
-		var/datum/effect/system/spark_spread/s = new /datum/effect/system/spark_spread
-		s.set_up(3, 1, M)
-		s.start()
-	if(d >= braindamage_start && prob(33))
-		M.adjustBrainLoss(1)
-	..()
+	var/d = data
+	if(M.get_species() == "Machine")
```
Comment (Crazylemon64) [inline code/modules/reagents/newchem/drinks.dm]: owwwww

See if you can use `isSynthetic` instead

**Comment by FalseIncarnate** [comment]
I'd put a check into the on_mob_life for organics and have it completely skip all effects beyond the vomit chance if you don't want people being drunk off a drink they can't actually process.

Also not sure how I feel about the holy water glass, it's probably best to leave it showing as water as it shows up as water in things like the chem machine.

You really shouldn't be direct-uploading changed files to Github like that, and instead be using something like the Github desktop client or one of the many other alternatives. It might work okay now, but it's probably meant for new files rather than adjusted ones.

**Comment by monster860** [comment]
Your commit messages...

**Comment by monster860** [comment]
But yeah, dont use the online editor unless you are uploading from your phone at school, like me.

**Comment by Crazylemon64** [comment]
@FlattestGuitar I also agree with @FalseIncarnate in that the glass of holy water should be indistinguishable from a glass of normal water, as it is clear that it is supposed to be difficult to differentiate from the chemmaster and dispenser.

File: code/modules/reagents/reagent_containers/food/drinks/drinkingglass.dm [hunk 2]
```diff
@@ -580,6 +580,22 @@
...
+					name = "Glass of Synthignon"
+					desc = "Someone mixed good wine and robot booze. Romantic, but atrocious."
+				if("uplink")
+					icon_state = "uplinkglass"
+					name = "Glass of Uplink"
+					desc = "An exquisite mix of the finest liquoirs and synthanol. Meant only for synthetics."
+				if("holywater")
+					icon_state = "glass_clear"
+					name = "Glass of Holy Water"
+					desc = "The contents of this glass seem to be filled with an energy of some kind."
```
Comment (Crazylemon64) [inline code/modules/reagents/reagent_containers/food/drinks/drinkingglass.dm]: This should probably just be the name and desc of a glass of normal water, so it doesn't show up as a glass of "what".

**Comment by NTSAM** [comment]
:+1:

**Comment by FlattestGuitar** [comment]
@FalseIncarnate the `on_mob_life` is already under ethanol and what's in here only expands on that. I don't see any clean way of limiting its effects on only organics. Currently it's possible to pour "water" into a glass and if you see it turn into a "Glass of.. what?" that means it's holy water.

As for the good practices of GitHub usage... as long as it works it's good, right? heh, heheh, heee..... I'll go hang myself.

File: code/modules/reagents/newchem/drinks.dm [hunk 3]
```diff
@@ -93,16 +94,32 @@
...
+		..()
+	else
+		if(prob(8))
+			M.fakevomit()
+
+
+datum/reagent/ethanol/synthanol/reaction_mob(var/mob/M, var/method=TOUCH, var/volume)
+	if(!istype(M, /mob/living))
+		return
+	if(M.get_species()=="Machine")
```
Comment (Fox-McCloud) [inline code/modules/reagents/newchem/drinks.dm]: spacing

File: code/modules/reagents/reagent_containers/food/drinks/drinkingglass.dm (hunk 2 above)
Comment (Fox-McCloud) [inline code/modules/reagents/reagent_containers/food/drinks/drinkingglass.dm]: Holy water is typically meant to be indistinguishable from regular water, so I'd have this hat the same description as water does.

**Comment by monster860** [comment]
Also, since when does ss13 have "clean coding practices?" This game was decompiled.

File: code/modules/reagents/reagent_containers/food/drinks/drinkingglass.dm [hunk 4]
```diff
@@ -580,6 +580,22 @@
...
+					name = "Glass of Synth 'n Soda"
+					desc = "Classic drink altered to fit the tastes of a robot. Bad idea to drink if you're made of carbon."
+				if("synthignon")
+					icon_state = "synthignonglass"
+					name = "Glass of Synthignon"
+					desc = "Someone mixed good wine and robot booze. Romantic, but atrocious."
+				if("uplink")
+					icon_state = "uplinkglass"
+					name = "Glass of Uplink"
+					desc = "An exquisite mix of the finest liquoirs and synthanol. Meant only for synthetics."
```
Comment (KasparoVy) [inline code/modules/reagents/reagent_containers/food/drinks/drinkingglass.dm]: "Liquoirs" to "liquors" or "liqueurs"

File: code/modules/reagents/reagent_containers/food/drinks/drinkingglass.dm [hunk 5]
```diff
@@ -580,6 +580,22 @@
 					icon_state = "servoglass"
 					name = "Glass of Servo"
 					desc = "Chocolate - based drink made for IPCs. Not sure if anyone's actually tried out the recipe."
+				if("synthnsoda")
+					icon_state = "synthnsodaglass"
+					name = "Glass of Synth 'n Soda"
+					desc = "Classic drink altered to fit the tastes of a robot. Bad idea to drink if you're made of carbon."
```
Comment (KasparoVy) [inline code/modules/reagents/reagent_containers/food/drinks/drinkingglass.dm]: "a carbon based life-form" maybe

**Comment by monster860** [comment]
Ipcs technically have carbon. Plasteel -> Steel -> Has carbon. Now we need silicon life forms. The natural kind.

**Comment by KasparoVy** [comment]
But they're not alive in the organic sense, they are sentient

File: code/modules/reagents/reagent_containers/food/drinks/drinkingglass.dm (hunk 4 above)
Comment (FlattestGuitar) [inline code/modules/reagents/reagent_containers/food/drinks/drinkingglass.dm]: _commits sepuku_

**Comment by TheDZD** [comment]
For the love of god, [set up Git properly](https://tgstation13.org/wiki/Setting_up_git), having one commit per file you added **and** using the web editor/Github for Windows client is horrendous.

**Comment by FlattestGuitar** [comment]
Thanks, @TheDZD.

**Comment by DavidRossii** [comment]
Regarding this point here - Holy Water have its own glass (uses the boring sprite of a water glass though). -

I'm not so sure about this honestly, the idea that holywater can now be identifiable is kinda lame for Cult rounds, sneaking holy water into their drinks etc.

I'm cool with it being identifiable in a chem-master, but just looking at it shouldn't be able to be identified as holy water.

Also how do you identify the difference between holy water and just normal water with plain eyes? - "An aura surrounds this glass, as if its contents were filled with an energy of some kind." - :+1: for the IPC guff, :-1: for the identification of holy water :disappointed:

**Comment by FlattestGuitar** [comment]
I am literally the worst. Commit is supposed to say holy water is indeed indistinguishable.

As for what @FalseIncarnate said: ethanol has its own `on_mob_life()` and synthanol is its child, so it uses both ethanol's and its own `on_mob_life()`.

I have absolutely no idea how to make the slurring not work for organics other than rip synthanol out from under ethanol. Obviously any help appreciated.

**Comment by Fox-McCloud** [comment]
I'm not going to merge something in that has a bunch of commits because you edited it with the web-editor.

Things need to be compiled and tested.

**Comment by FlattestGuitar** [comment]
I literally just got my stuff straight and set it up properly on my desktop.

The lates commit has been compiled and tested and uploaded from desktop.

**Comment by Fox-McCloud** [comment]
I realize that, but those first few commits are just...plain unacceptable.

File: code/modules/reagents/newchem/drinks.dm [hunk 6]
```diff
@@ -93,16 +94,32 @@
...
-
-	if(d >= collapse_start && prob(10))
-		M.emote("collapse")
-		var/datum/effect/system/spark_spread/s = new /datum/effect/system/spark_spread
-		s.set_up(3, 1, M)
-		s.start()
-	if(d >= braindamage_start && prob(33))
-		M.adjustBrainLoss(1)
-	..()
+	var/d = data
```
Comment (Iamgoofball) [inline code/modules/reagents/newchem/drinks.dm]: do NOT use data holy FUCK 

data is B A D

use the dedicated "how many times have I ticked" var holy FUCK BALLS SHIT AAAAAAAA

Comment (Fox-McCloud) [inline code/modules/reagents/newchem/drinks.dm]: Ethanol already works this way---I don't expect him to refactor it, but yeah, data is badddd.

**Comment by FlattestGuitar** [comment]
Okay, any ideas for making organics not slur when drinking synthanol? I could just make it not be a child of ethanol. Considering there's a lot more differences now it wouldn't even be that big of a copypasta.

**Comment by VampyrBytes** [comment]
How about, instead of repathing it, increase the vomit chance, and when they throw up remove some/all the synthanol from their system... after all, they just threw it up. That way they won't be keeping enough down to start slurring.

Not keen on repathing as it is still a type of ethanol, it's just a type designed for machines

**Comment by FlattestGuitar** [comment]
I'll make a new PR once I figure out GitHub and work on a few more things about synthanol.

**Pull request closed without merging**